cmd/dump.go

```diff
@@ -0,0 +1,285 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/yaml"
+
+	kuadrantv1alpha1 "github.com/kuadrant/kuadrant-operator/api/v1alpha1"
+	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	kuadrantv1beta2 "github.com/kuadrant/kuadrant-operator/api/v1beta2"
+)
+
+var (
+	dumpNamespace     string
+	dumpAllNamespaces bool
+	dumpOutputDir     string
+	dumpFormat        string
+)
+
+func dumpCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dump",
+		Short: "Dump Kuadrant and Gateway API resources",
+		Long:  "Dump all Kuadrant and Gateway API resources to files for investigation and debugging",
+		RunE:  runDump,
+	}
+
+	cmd.Flags().StringVarP(&dumpNamespace, "namespace", "n", "", "Namespace to dump resources from (default: all namespaces)")
+	cmd.Flags().BoolVarP(&dumpAllNamespaces, "all-namespaces", "A", false, "Dump resources from all namespaces")
```

Comment:
Can there be some input validation on the inputs. The user should not be able to set the namespace and all-namespaces at the same time. 

I also don't think the default should be to dump all the namespaces if not set.